install/components/upgrade-php/upgrade-php.sh

```diff
@@ -0,0 +1,193 @@
+	exit 1
+fi
+
+
+# If no args were passed, script_echo out some helpful info
+if [ $# -eq 0 ]; then
+	script_echo "Usage: upgrade-php.sh OLD_VERSION NEW_VERSION"
+	script_echo "Example: upgrade-php.sh 8.0 8.2"
+	script_echo "Notes:"
+	script_echo '- Set $DEBUG for a verbose output - $DEBUG=true ./upgrade-php.sh [options]'
+	exit 1
+fi
+
+# For debugging the script
+# set -v if
+if [ ! -z "$DEBUG" ]; then
+	set -v
+fi
+
+# Validate that the user passed at least two args
+if [ $# -lt 2 ]; then
+	script_echo "Please pass the version args (OLD_VERSION NEW_VERSION)"
+	script_echo "Example: upgrade-php.sh 8.0 8.2"
+	exit 1
+fi
+
+# Validate that the version args are valid
+# they should be numbers with a decimal point
+if [[ ! $1 =~ ^[0-9]+(\.[0-9]+)?$ ]] || [[ ! $2 =~ ^[0-9]+(\.[0-9]+)?$ ]]; then
+	script_echo "Please pass valid version args"
+	script_echo "Example: upgrade-php.sh 8.0 8.2"
+	exit 1
+fi
+
+OLD_PHP_VERSION=$1
+NEW_PHP_VERSION=$2
+
+# Install new versions of PHP and php-fpm but only if they don't already exist
+if [ -f /usr/bin/php${NEW_PHP_VERSION} ]; then
+	script_echo "ℹ️ php${NEW_PHP_VERSION} already exists, skipping install"
+else
+	script_echo "📦 Installing php${NEW_PHP_VERSION}"
+	sudo apt-get update
+	sudo apt-get -y --no-install-recommends install php${NEW_PHP_VERSION}
+fi
+
+if [ -f /usr/sbin/php-fpm${NEW_PHP_VERSION} ]; then
+	script_echo "ℹ️ php-fpm${NEW_PHP_VERSION} already exists, skipping install"
+else
+	script_echo "📦 Installing php-fpm${NEW_PHP_VERSION}"
+	sudo apt-get update
+	sudo apt-get -y --no-install-recommends install php${NEW_PHP_VERSION}-fpm
+fi
+
+# Install all php packages if they don't already exist
+script_echo "📦 Installing php packages"
+sudo apt-get update && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-curl && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-mysql && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-dev && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-zip && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-mbstring && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-memcached && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-pgsql && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-xml && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-intl && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-redis && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-bcmath && \
+	sudo apt-get -y install php${NEW_PHP_VERSION}-gd
+
+# make edits to PHP.ini file
+script_echo "🖊 Editing php.ini"
+sed -i 's/;cgi.fix_pathinfo=1/cgi.fix_pathinfo=0/g' /etc/php/${NEW_PHP_VERSION}/fpm/php.ini
+
+# Enable all php extensions
+script_echo "📦 Enabling all php extensions"
+phpenmod -v $NEW_PHP_VERSION mbstring
+phpenmod -v $NEW_PHP_VERSION curl
+phpenmod -v $NEW_PHP_VERSION xml
+phpenmod -v $NEW_PHP_VERSION xmlreader
+phpenmod -v $NEW_PHP_VERSION simplexml
+phpenmod -v $NEW_PHP_VERSION gd
+
+# TODO: Why are there not more of these to enable? Compared to number that we installed
```

Comment:
im not sure. prob should enable them.

Reply:
confirmed that they're enabled by other means `php -m`